### ISSUE TITLE
Suggestion only: UI preventing instancing a child composite that is the same type as its parent composite

### ISSUE THREAD

**Issue by Const1702** [MattFiler/OpenCAGE#214]
I was experimenting with composites and wanted to see what would happen if the child composite was the same as its parent one. In my case I tried expanding the CAGEONLY\FRONT_MARKER, because there are no entities in it and makes it a simpler test subject. Then I added composite **CAGEONLY\FRONT_MARKER** inside of it and the editor either crashes or freezes and then crashes. That makes sense, but for new users that don't fully understand how composites work, it would be better to have some sort of UI like a message box letting them know that the child composite cannot be the same as its parent.

Level I did this on: **HAB_AIRPORTtest** (I cloned the HAB_AIRPORT level just in case, so it doesn't do permanent damage to the original one)

https://github.com/MattFiler/OpenCAGE/assets/116276106/70e27e8f-4a6f-4252-81e1-c2699bf35bef

_You were absent for some time, so I hope you got a very good rest! Thank you for working so hard on this project!_

**Comment by MattFiler** [comment]
Good spot, this is a logic error that shouldn't be possible.

> You were absent for some time, so I hope you got a very good rest! Thank you for working so hard on this project!

Thanks! I was away in Croatia, very nice country!

**Comment by Const1702** [comment]
Experimented a bit and noticed that OpenCAGE only detects when a composite is instanced directly in itself. If you try and do that in a composite's child composite, the editor will allow it and lead to a crash.

**Prevention works for a case like this:**
Composite_A -> Composite_A

**Prevention doesn't work for a case like this:**
Composite_A -> Composite_B -> Composite_A

For example I tried to instance `Door_Package` into `CuttingPanel_Empty`. Since `CuttingPanel_Empty` already exists inside `Door_Mechanism_Front` which exists in `Door_Package` having a parent composite inside of a child will lead to a crash. The editor did manage to last for some time and I tried following the entities through and got this path:
![Image](https://github.com/user-attachments/assets/5dc23618-8f42-4600-8266-88e63776743e)

When I took the screenshot it crashed.

I think, however, preventing the user from instancing a composite in its children would be difficult and might slow down performance. Creating a composite would need to check all composites on the level and their structure and I'm not sure if it is even necessary. I highly doubt that someone will instance a whole ENVIRONMENT_... inside of a door.
